bootstrap/peer.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package bootstrap
+
+import (
+	"context"
+	"net"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/server"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// DB is the master database for Boostrap Node
+type DB interface {
+	// TODO: use better interfaces
+	RoutingTable() (kdb, ndb storage.KeyValueStore)
+}
+
+// Config is all the configuration parameters for a Bootstrap Node
+type Config struct {
+	Server   server.Config
+	Kademlia kademlia.Config
+}
+
+// Verify verifies whether configuration is consistent and acceptable.
+func (config *Config) Verify(log *zap.Logger) error {
+	return config.Kademlia.Verify(log)
+}
+
+// Peer is the representation of a Bootstrap Node.
+type Peer struct {
+	// core dependencies
+	Log      *zap.Logger
+	Identity *identity.FullIdentity
+	DB       DB
+
+	// TODO: add transport
+
+	// servers
+	Public struct {
+		Listener net.Listener
+		Server   *server.Server
+	}
+
+	// services and endpoints
+	// TODO: similar grouping to satellite.Peer
+	RoutingTable      *kademlia.RoutingTable
```

Comment:
Here you can use structs similarly to how satellite does:
https://github.com/storj/storj/pull/1119/files#diff-e18cb5f53ca9336df150764f9fc70167R105 (ignore kdb/ndb thing there).